src/Command/RedisFactory.php

```diff
@@ -25,15 +27,18 @@
 class RedisFactory extends Factory
 {
     /**
-     *
+     * @var CommandResolverInterface
      */
-    public function __construct()
+    private $commandResolver;
+
+    public function __construct(CommandResolverInterface $commandResolver)
```

Comment:
Logic of command resolving becomes more complex since we adding support for modules, so In my opinion to avoid violation of SRP it makes sense to separate logic of object creation (factory) and resolving object from command ID (resolver)

Reply:
We can't break the interface in a point release, we can make this a `3.0` and release when it's feature complete with Redis Stack support?

Reply:
Makes sense to me. @chayim What do you think?